File: internal/models/product.go
package models

type ProductCreate struct {
	CategoryID      int    `json:"category_id" binding:"required"`
	Name            string `json:"name" binding:"required"`
	Characteristics string `json:"characteristics" binding:"required"`
}

type ProductRetrieve struct {
	ID              int    `json:"id"`
	CategoryID      int    `json:"category_id"`
	Name            string `json:"name"`
	Characteristics string `json:"characteristics"`
}

type ProductUpdate struct {
	CategoryID      *int    `json:"category_id"`
	Name            *string `json:"name"`
	Characteristics *string `json:"characteristics"`
}

File: frontend/src/types/product.ts
export interface Product {
  product_id: number;
  name: string;
  characteristics: string;
  category_id: number;
}

export interface ProductCreate {
  name: string;
  characteristics: string;
  category_id: number;
}

export interface ProductUpdate {
  name?: string;
  characteristics?: string;
  category_id?: number;
}
